src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepExecution.java

```diff
@@ -167,15 +173,37 @@ protected Job asJob() {
                             allParameters.put(defaultP.getName(), defaultP);
                         }
                     } else {
+                        String description = Util.fixNull(pDef.getDescription());
                         if (pDef instanceof ChoiceParameterDefinition) {
                             ParameterValue pv = allParameters.get(pDef.getName());
                             if (!((ChoiceParameterDefinition)pDef).getChoices().contains(pv.getValue())) {
                                 throw new AbortException("Value for choice parameter '" + pDef.getName() + "' is '" + pv.getValue() + "', "
                                         + "but valid choices are " + ((ChoiceParameterDefinition)pDef).getChoices());
                             }
+                        } else if (pDef instanceof StringParameterDefinition) {
```

Comment:
FWIW I meant to not even special-case passwords at all, just handle all `SimpleParameterDefinition`s the same way, something like this:

```
if (pDef instanceof SimpleParameterDefinition && !pDef instanceof StringParameterDefinition) {
    ParameterValue pv = allParameters.get(pDef.getName());
    if (pv instanceof StringParameterValue) {
        String desiredValue = (String) pv.getValue();
        listener.getLogger().println(String.format("The types of the specified parameters did not match the expected types defined on the target job. Converting '%s' to satisfy '%s'.", pv.getName(), pDef.getClass().getSimpleName()));
        ParameterValue convertedValue = ((SimpleParameterDefinition) pDef).createValue(desiredValue);
        allParameters.put(pDef.getName(), convertedValue);
    }
}
```
To me that seems ideal because it means that pipeline-build-step and parameterized-trigger have the exact same behavior, which seems least likely to cause confusion.

I like the idea of throwing `AbortException` if the parameter types do not match and cannot be converted, but parameterized-trigger does not seem to do that, so I am a little hesitant to be more strict here, and I have no idea of whether that case would occur in practice and might break things, or if it should never happen at all.

What do you think? Not trying to force one way or the other, just trying to understand the problem space better.

Reply:
Also, let me know if you want me to take this over, not sure how much time you have to spend on it.

Reply:
It was just to speed-up the reaction by providing the initial investigation, unit tests and a potentially working solution. So if you can take over the task, it will be with pleasure :) 

Concerning the special cases, it's because the parameterized-trigger and this plugin does not share the same situation IIUC. Within a pipeline, you cannot pass any arbitrary parameter, like `choiceParam` are passed as `string`. Perhaps it's just due to my minimum knowledge of the context.

The only case that *must* be covered is when you pass a string and the job expected a password, all the other behaviors are up to you ;)